common/compiler-input.ts

```diff
@@ -0,0 +1,261 @@
+  /** Version of the EVM to compile for. Affects type checking and code generation */
+  evmVersion: EvmVersion;
+  /**
+   * Change compilation pipeline to go through the Yul intermediate representation.
+   * This is a highly EXPERIMENTAL feature, not to be used for production. This is false by default.
+   */
+  viaIR?: boolean
+  /** Debugging settings */
+  debug?: CompileDebug
+  /** Metadata settings */
+  metadata?: CompilerMetadata
+  /** Addresses of the libraries. If not all libraries are given here, it can result in unlinked objects whose output data is different. */
+  libraries: CompilerLibraries
+  /**
+   * The following can be used to select desired outputs.
+   * If this field is omitted, then the compiler loads and does type checking, but will not generate any outputs apart from errors.
+   * The first level key is the file name and the second is the contract name, where empty contract name refers to the file itself,
+   * while the star refers to all of the contracts.
+   * Note that using a using `evm`, `evm.bytecode`, `ewasm`, etc. will select every
+   * target part of that output. Additionally, `*` can be used as a wildcard to request everything.
+   * Contract level (needs the contract name or "*"):
+   * - abi - ABI
+   * - devdoc - Developer documentation (natspec)
+   * - userdoc - User documentation (natspec)
+   * - metadata - Metadata
+   * - ir - Yul intermediate representation of the code before optimization
+   * - irOptimized - Intermediate representation after optimization
+   * - storageLayout - Slots, offsets and types of the contract's state variables.
+   * - evm.assembly - New assembly format
+   * - evm.legacyAssembly - Old-style assembly format in JSON
+   * - evm.bytecode.functionDebugData - Debugging information at function level
+   * - evm.bytecode.object - Bytecode object
+   * - evm.bytecode.opcodes - Opcodes list
+   * - evm.bytecode.sourceMap - Source mapping (useful for debugging)
+   * - evm.bytecode.linkReferences - Link references (if unlinked object)
+   * - evm.bytecode.generatedSources - Sources generated by the compiler
+   * - evm.deployedBytecode* - Deployed bytecode (has all the options that evm.bytecode has)
+   * - evm.deployedBytecode.immutableReferences - Map from AST ids to bytecode ranges that reference immutables
+   * - evm.methodIdentifiers - The list of function hashes
+   * - evm.gasEstimates - Function gas estimates
+   * - ewasm.wast - Ewasm in WebAssembly S-expressions format
+   * - ewasm.wasm - Ewasm in WebAssembly binary format
+   */
+  outputSelection?: CompilerOutputSelection
+  /** The modelChecker object is experimental and subject to changes. */
+  modelChecker?: CompilerModelChecker
+}
+
+export interface CompilationInput {
+  /** Source code language */
+  language: 'Solidity' | 'yul'
+  sources: SourcesInput
+  settings?: CompilerSettings
+}
+
+export interface CondensedCompilationInput {
+  language: 'Solidity' | 'Vyper' | 'lll' | 'assembly' | 'yul'
```

Comment:
What is this based on? Technically, [solc supports only `Solidity` and `Yul` (case-sensitive)](https://github.com/ethereum/solidity/blob/a53f15f45f9089623ec91961602f6a07ff0f086e/libsolidity/interface/StandardCompiler.cpp#L1514-L1519). In a broader context that also includes any compilers supporting Standard JSON I'd say that the values are not limited to some predefined choices - other compilers are free to support any language identifiers.